src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -39,6 +39,7 @@ public sealed class WebAssemblyHostBuilder
         /// <returns>A <see cref="WebAssemblyHostBuilder"/>.</returns>
         [DynamicDependency(nameof(JSInteropMethods.NotifyLocationChanged), typeof(JSInteropMethods))]
         [DynamicDependency(JsonSerialized, typeof(WebEventDescriptor))]
+        [DynamicDependency(Component, typeof(HeadOutlet))]
```

Comment:
Could I bother you to

a) Add the comment from your PR here?
b) Possibly update the WebAssembly pre-rendering scenario to include the HeadComponent?

The interesting thing is that the WebAssembly pre-rendering would run in to the same issue with all components in the framework if they happen to be root components. I can think of too many that would qualify for one, but it's something to be cognizant of when designing new ones. 


FYI @javiercn / @SteveSandersonMS

Reply:
Since this type of bug could be possible with any arbitrary root component not referenced in the client-side build, I wonder if it could be worth it to throw a meaningful exception when a prerendered component type could not be found in its assembly. If we come up with an intuitive workaround, the exception message could also provide info about that.

Reply:
>  if it could be worth it to throw a meaningful exception when a prerendered component type could not be found in its assembly

That seems like a good idea. We wouldn't have spent that long scratching our heads if things had failed noisily. @javiercn how do you feel about this? I imagine the change would be in the code that rehydrates from WebAssembly pre-rendering.

Reply:
Agreed it would be good if we can do that.

We can afford a bit of extra overhead (for example, explicitly calling `Type.GetType` or whatever to see if we know about the type) in the code that rehydrates since it's a one-time cost during startup rather than being part of the core rendering loop.